substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/PointsToAnalyzer.java

```diff
@@ -0,0 +1,395 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ForkJoinPool;
+
+import static org.graalvm.compiler.replacements.StandardGraphBuilderPlugins.registerInvocationPlugins;
+
+public final class PointsToAnalyzer {
+
+    static {
+        ModuleSupport.exportAndOpenAllPackagesToUnnamed("jdk.internal.vm.ci", false);
+        ModuleSupport.exportAndOpenAllPackagesToUnnamed("jdk.internal.vm.compiler", false);
+        ModuleSupport.exportAndOpenAllPackagesToUnnamed("jdk.internal.vm.compiler.management", true);
+        ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "jdk.internal.loader", false);
+        ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "jdk.internal.misc", false);
+        if (JavaVersionUtil.JAVA_SPEC >= 15) {
+            ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "jdk.internal.misc", false);
+        }
+        ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "sun.text.spi", false);
+        ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "jdk.internal.org.objectweb.asm", false);
+        if (JavaVersionUtil.JAVA_SPEC >= 16) {
+            ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "sun.reflect.annotation", false);
+            ModuleSupport.exportAndOpenPackageToUnnamed("java.base", "sun.security.jca", false);
+            ModuleSupport.exportAndOpenPackageToUnnamed("jdk.jdeps", "com.sun.tools.classfile", false);
+        }
+    }
+
+    private OptionValues options;
+    private final StandalonePointsToAnalysis bigbang;
+    private AnalysisFeatureManager analysisFeatureManager;
+    private AnalysisClassLoader analysisClassLoader;
+    private final DebugContext debugContext;
+    private AnalysisFeatureImpl.OnAnalysisExitAccessImpl onAnalysisExitAccess;
+    private String analysisName;
+    private boolean entrypointsAreSet;
+    private boolean mainEntryIsSet;
+
+    @SuppressWarnings("try")
+    private PointsToAnalyzer(OptionValues options) {
+        this.options = options;
+        analysisFeatureManager = new AnalysisFeatureManager(options);
+        String appCP = PointstoOptions.AnalysisTargetAppCP.getValue(options);
+        if (appCP == null) {
+            AnalysisError.shouldNotReachHere("Must specify analysis target application's classpath with -H:" + PointstoOptions.AnalysisTargetAppCP.getName());
+        }
+        List<URL> urls = new ArrayList<>();
+        for (String cp : appCP.split(File.pathSeparator)) {
+            try {
+                File file = new File(cp);
+                if (file.exists()) {
+                    urls.add(file.toURI().toURL());
+                }
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            }
+        }
+
+        analysisClassLoader = new AnalysisClassLoader(urls.toArray(new URL[0]), ClassLoader.getPlatformClassLoader());
+        Providers providers = getGraalCapability(RuntimeProvider.class).getHostBackend().getProviders();
```

Comment:
You could access the original providers through `Providers originalProviders = GraalAccess.getOriginalProviders();`.